### PR TITLE
Mysql fix prepared statement #2705

### PR DESCRIPTION
**Pull request by lewie**
@watou: You're welcome...

### CODE REVIEW

**Comment by lewie** [comment]
@teichsta 
But this one works well... :-)

**Comment by teichsta** [comment]
could @watou or @kabili207 confirm this fix is working? Thank, Thomas E.-E.

**Comment by kabili207** [comment]
Looks good to me. Previously problematic items are now being persisted correctly.

File: bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java [hunk 1]
```diff
@@ -117,7 +119,7 @@ public void activate() {
 		sqlTypes.put("CONTACTITEM", "VARCHAR(6)");
 		sqlTypes.put("DATETIMEITEM", "DATETIME");
 		sqlTypes.put("DIMMERITEM", "TINYINT");
-		sqlTypes.put("GROUPITEM", "DOUBLE");
+		sqlTypes.put("GROUPITEM", "VARCHAR(200)");
```
Comment (watou) [inline bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: Is this change supposed to be part of this pull request?

Comment (lewie) [inline bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: No, it isn't! Both are wrong. With my other PR Type tests for groups will be introduced.

Comment (teichsta) [inline bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: hmm … then this change shouldn't look different, right?

**Comment by teichsta** [comment]
please also squash your commits into one finally. Thanks, Thomas E.-E.

**Comment by lewie** [comment]
Hello @teichsta, here I have tried to squash:
https://github.com/openhab/openhab/pull/2824
I'm not sure that's right now?

Rgds
Helmut

**Comment by watou** [comment]
[Here](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit) is one person's suggestion on how to squash commits in pull requests.  When the top of your pull requests shows "Commits 1", you've done it!

**Comment by lewie** [comment]
@watou, Thanks

**Comment by lewie** [comment]
After some more Git Lessons, I have tried to squash directly now, but always got a Merge branch added automatically.  
Not to do a Pull after rebase and --force in my Push Command did the Trick.  (@watou, Thanks again!)

Rgds
Helmut

**Comment by watou** [comment]
@lewie, glad to help!

**Comment by teichsta** [comment]
i still see five commits … :-)

**Comment by lewie** [comment]
MIST!
"This branch is 5 commits ahead, 63 commits behind openhab:master"

3 of them are my "Merge pull requests" to update my main Fork. :-( 
Currently I have 3 PRs for OpenHab. Puh, how can I stop that Chaos?

**Comment by lewie** [comment]
@teichsta, hope this is right now, went back to current master.
Use this PR please:  https://github.com/openhab/openhab/pull/2831
When this is working right, I will push the other "clean" Branches.

Thank you for your patience
Helmut

**Pull request closed without merging**